### PR TITLE
uses argparse to read in values from commandline.

### PR DESCRIPTION
**Pull request by bsubei**
figured I'd make an improved version of #1 to demonstrate [argparse](https://docs.python.org/2/library/argparse.html)

### CODE REVIEW

**Comment by bsubei** [comment]
fair warning: I haven't tested this on a set of images (would be great if someone could post a small set : )

**Comment by heyitswither** [comment]
+1

**Comment by arguiot** [comment]
+1

File: rollingshutter.py [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
     
     return 'DONE'
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input_directory', required=True, help='Path to directory containing input images.')
+    parser.add_argument('-o', '--output-directory', required=True, help='Path to directory to write output image(s) to.')
+    parser.add_argument('-s', '--speed', default=1, type=int, help='Speed variable to control how many rows to jump by per frame.')
+    parser.add_argument('-x', '--extension-type', default='png', help='Speed variable to control how many rows to jump by per frame.')
```
Comment (RoelAdriaans) [inline rollingshutter.py]: The help for the extention type is the same as for the speed

**Review by RoelAdriaans — CHANGES_REQUESTED** [review]

File: rollingshutter.py [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
+    if not os.path.exists(output_directory):
+            os.makedirs(output_directory)
+    output_image.save(os.path.join(output_directory, 'output_image.png'))
     
     return 'DONE'
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input_directory', required=True, help='Path to directory containing input images.')
```
Comment (RoelAdriaans) [inline rollingshutter.py:33]: Maybe do not use named arguments, but positional. 
Most linux utils, like cp, ls,... use those for filenames. 

Then you can do rollingshutter.py in/ out/, makes is a bit cleaner on the console?

File: rollingshutter.py [hunk 3]
```diff
@@ -1,31 +1,46 @@
...
+    output_image = Image.new('RGB', (image_width, image_height)) 
     
     
     # let us go through the frames one at a time
     
     current_row = 0
-    
-    for filename in glob.glob(frame_dir + "*." + frame_file):
+     
+    for filename in glob.glob(os.path.join(input_directory, "*.", extension_type)):
```
Comment (colcomb) [inline rollingshutter.py]: `os.path.join(input_directory, "*.", extension_type)` adds a slash between the files and the extension.
Should be `os.path.join(input_directory, "*." + extension_type)`.

Comment (bsubei) [reply rollingshutter.py]: d'oh! Nice catch!

File: rollingshutter.py (hunk 2 above)
Comment (bsubei) [reply rollingshutter.py:33]: I dunno, "explicit is better than implicit" is usually my approach (I don't want to remember the order of arguments). Either way, I don't think it matters too much.

File: rollingshutter.py (hunk 1 above)
Comment (bsubei) [reply rollingshutter.py]: woops

**Comment by bsubei** [comment]
closed due to inactivity

**Pull request closed without merging**